docs/development/component-checklist.md

```diff
@@ -125,6 +125,12 @@ This document provides a checklist for them that you can walk through.
 
    <sub>[*] It is required because if a component deployed in the Shoot cluster does not specify a Seccomp profile and cannot run with the `RuntimeDefault` Seccomp profile, then enabling the `.spec.kubernetes.kubelet.seccompDefault` field in the Shoot spec would break the corresponding component.</sub>
 
+9. **Use strict mode when decoding data** ([example](https://github.com/gardener/gardener/blob/485c25124ea536e4610c627109017dd34d434921/pkg/provider-local/controller/worker/actuator.go#L52))
```

Comment:
Shouldn't this be part of (sub-bullet in) a bigger topic of "Add input validation" - make sure that you add validation for your API?
If we have other guidelines and recommendations how to do input validation, we could even write a dedicated guide and link it from the component checklist.